common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -1450,5 +1450,9 @@ def default(self, obj):
                     return obj.isoformat()
             else:
                 return obj.isoformat()
+        # The type named '__proxy__' is meant to be the one in (django.utils.functional). We cannot
+        # use (isinstance) directly for comparison because this module is loaded before (django)
+        elif type(obj).__name__=='__proxy__':
```

Comment:
I tried that. Once I import **`django`**, then system crashes. This means that this module is initialized before django (as far as I can understand from that!)

Reply:
You could do a local import (inside the function, rather than at the module level), which would delay it until the serializer is first used. You could also catch the import error, and simply exclude that check if django is not available.

Reply:
Thank you @cpennington . I tired the local import from django.utils.functional import Promise and I got an error No module named utils.functional . I tired the try ... except at the module level, but the import never succeeded. In both solutions, the system immediately crashes !

Reply:
I agree that a local import (with a comment) is the best solution here.  I don't understand why the django import wouldn't work after line 1442.

Reply:
Moving the **'EdxJSONEncoder'** class out of this file did the trick :+1: